storage/minio/src/server/MinioServerStorage.ts

```diff
@@ -26,4 +46,37 @@ export class MinioServerStorage extends S3ServerStorage {
     if (!(await this.baseDirectoryExists(reference)))
       await this.createBaseDirectory(reference);
   }
+
+  private replaceSymbols(textToReplace: string) {
+    const { symbolsMap } = this._config;
+    let result = textToReplace;
+    for (const key in symbolsMap) {
+      if (symbolsMap.hasOwnProperty(key))
+        result = result.replace(new RegExp(key, "g"), symbolsMap[key]);
+    }
+    return result;
+  }
+
+  private async replaceReferenceSymbols(
+    reference: ObjectReference
+  ): Promise<ObjectReference> {
+    const result = reference;
+    result.baseDirectory = this.replaceSymbols(reference.baseDirectory);
+    result.objectName = this.replaceSymbols(reference.objectName);
+    if (reference.relativeDirectory)
+      result.relativeDirectory = this.replaceSymbols(
+        reference.relativeDirectory
+      );
+    return result;
+  }
+
+  public override async getUploadUrl(
```

Comment:
What about other methods that write data to storage and accept a directory name of some sort:
`upload`
`uploadInMultipleParts`
`createBaseDirectory`

@Animagne would be interesting to hear your thoughts on this.
Current solution alters the user-passed object to set the normalized path of the uploaded file
![image](https://github.com/iTwin/object-storage/assets/70565417/d2b5d441-8f95-4192-8b2b-65d7f9ed76b5)
To me this looks like an anti-pattern however it looks like we have no easy way to return this information back to user. 
Should we consider returning an `ObjectReference` object from all the methods? In most of the cases we would just return whatever user passed to us but in a case such like this we could construct a new object with correct file paths.

Last week we briefly discussed an interface change request raised by Paulius, maybe this could go into the same major version.